cabot/cabotapp/tasks.py

```diff
@@ -139,3 +154,50 @@ def clean_db(days_to_retain=60):
 
     clean_db.apply_async(kwargs={'days_to_retain': days_to_retain},
                          countdown=3)
+
+
+@task(ignore_result=True)
+def send_schedule_problems_email(schedule_id):
+    """
+    Send off an email as a celery task
+    :param schedule_id schedule ID
+    """
+    try:
+        schedule = models.Schedule.objects.get(pk=schedule_id)
+        problems = schedule.problems
+    except models.Schedule.DoesNotExist, models.ScheduleProblems.DoesNotExist:
+        # if the schedule or problems got deleted, nothing to do
+        return
+
+    # check if problems became silenced since the email got queued
+    if problems.is_silenced():
+        return
+
+    # build the message
+    # make the schedule link absolute (add domain name) because this is going into an email
+    cabot_schedule_url = build_absolute_url(schedule.get_edit_url())
+
+    # build links to snooze email alerts
+    snooze_hours = [4, 12, 24]
+    snooze_links = [build_absolute_url(reverse('snooze-schedule-warnings',
+                                               kwargs={'pk': schedule.pk, 'hours': hours})) for hours in snooze_hours]
+    snoozes = ['<a href="{}">{} hours</a>'.format(link, hours) for link, hours in zip(snooze_links, snooze_hours)]
+
+    message = 'The schedule <a href="{}">{}</a> has some issues:\n\n{}\n\n' \
+              'Click <a href="{}">here</a> to review the schedule\'s configuration.\n' \
+              'If you don\'t want to deal with this right now, you can silence these alerts for {}.' \
+        .format(cabot_schedule_url, schedule.name, problems.text, cabot_schedule_url, ' | '.join(snoozes))
+
+    # figure out who to send it to (on-call + fallback)
+    recipients = models.get_duty_officers(schedule) + models.get_fallback_officers(schedule)
+    recipients = [r.email for r in recipients if r.email]
+
+    if len(recipients) > 0:
+        send_mail(subject="Cabot Schedule '{}' Has Problems".format(schedule.name),
+                  message=message,
+                  from_email='Cabot Updates<{}>'.format(os.environ.get('CABOT_FROM_EMAIL')),
+                  recipient_list=recipients)
+    else:
+        # if no one is available, log a warning as a last-ditch attempt to get someone's attention
```

Comment:
I would log a warning no matter what, for increased visibility.

Reply:
👍